extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerUtil.java

```diff
@@ -15,16 +15,21 @@ private TracerUtil() {
     }
 
     public static Resource mapResourceAttributes(List<String> resourceAttributes, String serviceName) {
+        // resource service name is always set at build time to the app name, therefore
```

Comment:
This is getting too hard to read... Would this work?
```
public static Resource mapResourceAttributes(List<String> resourceAttributes, String serviceName) {
    final boolean setServiceName = serviceName != null;
    final AttributesBuilder attributesBuilder = Attributes.builder();

    if (!resourceAttributes.isEmpty()) {
        var attributes = OpenTelemetryUtil.convertKeyValueListToMap(resourceAttributes);
        attributes.forEach(attributesBuilder::put);
    }

    if (setServiceName) {
        attributesBuilder.put(SERVICE_NAME.getKey(), serviceName);
    }

    return Resource.create(attributesBuilder.build());
}
```

Reply:
sure, will do